### ISSUE TITLE
non-applicable constraints must be ignored for remote tracks

### ISSUE THREAD

**Issue by kuddai** [w3c/mediacapture-main#619]
In the presence of  ["Clarify what happens to non-applicable constraints"](https://github.com/w3c/mediacapture-main/pull/542) and https://w3c.github.io/webrtc-pc/#mediatracksupportedconstraints-mediatrackcapabilities-mediatrackconstraints-and-mediatracksettings

> The basics of MediaTrackSupportedConstraints, MediaTrackCapabilites, MediaTrackConstraints and MediaTrackSettings is outlined in [GETUSERMEDIA]. However, the MediaTrackSettings for a MediaStreamTrack sourced by an RTCPeerConnection will only be populated with members to the extent that data is supplied by means of the remote RTCSessionDescription applied via setRemoteDescription and the actual RTP data. This means that certain members, such as facingMode, echoCancellation, latency, deviceId and groupId, will always be missing.

should we just ignore such members as latency or echoCancellation on remote audio tracks instead of throwing OverconstrainedError? By remote audio tracks I mean track which is produced by [RTCRtpReceiver](https://w3c.github.io/webrtc-pc/#dom-rtcrtpreceiver). 

The issue is equally applicable to [w3c.github.io/webrtc-pc](https://w3c.github.io/webrtc-pc/) as it occurs on spec intersection.

**Comment by jan-ivar** [comment]
This was [recently](https://github.com/w3c/mediacapture-main/pull/594) clarified by [this note](https://w3c.github.io/mediacapture-main/getusermedia.html#media-track-supported-constraints): *"The constraints specified in this specification apply only to instances of MediaStreamTrack generated by getUserMedia(), unless stated otherwise in other specifications."*

For example, [mediacapture-screen-share](https://w3c.github.io/mediacapture-screen-share/#constrainable-properties-for-captured-display-surfaces) specifies the constrainable properties of tracks from `getDisplayMedia()`, and [how](https://w3c.github.io/mediacapture-screen-share/#downscaling-and-frame-decimation) to implement them.

In other words, it is up to webrtc-pc to specify what constraints remote tracks have. This is being discussed in https://github.com/w3c/webrtc-pc/issues/2121.

> should we just ignore such members as latency or echoCancellation on remote audio tracks

FWIW, about those members specifically, [webrtc-pc](https://w3c.github.io/webrtc-pc/#mediatracksupportedconstraints-mediatrackcapabilities-mediatrackconstraints-and-mediatracksettings) already says: *"This means that certain members, such as facingMode, echoCancellation, latency, deviceId and groupId, will always be missing."*

**Comment by kuddai** [comment]
I wasn't specific enough. It is more like Chrome specific issue. On Firefox I got consistent behavior. If I try to apply "latency" constraint to either audio or video track on RTCRtpReceiver then I got OverconstrainedError with rejected promise.

On Chrome however, I will get OverconstrainedError for audio track and resolved promise for video track on RTCRtpReceiver.

**Comment by armax00** [comment]
For incosistent behavior on Chrome please file a bug on crbug.com.

Cheers,
A.

**Comment by kuddai** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=1007933
